Mathlib/Order/KrullDimension.lean

```diff
@@ -30,40 +40,106 @@ in this file would generalize as well. But we don't think it would be useful, so
 Krull dimension of a preorder.
 -/
 
-section definitions
+namespace Order
 
-variable (α : Type*) [Preorder α]
+section definitions
 
 /--
 The **Krull dimension** of a preorder `α` is the supremum of the rightmost index of all relation
-series of `α` order by `<`. If there is no series `a₀ < a₁ < ... < aₙ` in `α`, then its Krull
+series of `α` ordered by `<`. If there is no series `a₀ < a₁ < ... < aₙ` in `α`, then its Krull
 dimension is defined to be negative infinity; if the length of all series `a₀ < a₁ < ... < aₙ` is
 unbounded, its Krull dimension is defined to be positive infinity.
 -/
-noncomputable def krullDim : WithBot (WithTop ℕ) :=
+noncomputable def krullDim (α : Type*) [Preorder α] : WithBot ℕ∞ :=
   ⨆ (p : LTSeries α), p.length
 
 /--
-Height of an element `a` of a preordered set `α` is the Krull dimension of the subset `(-∞, a]`
+The **height** of an element `a` in a preorder `α` is the supremum of the rightmost index of all
+relation series of `α` ordered by `<` and ending with `a`.
 -/
-noncomputable def height (a : α) : WithBot (WithTop ℕ) := krullDim (Set.Iic a)
+noncomputable def height {α : Type*} [Preorder α] (a : α) : ℕ∞ :=
+  ⨆ (p : LTSeries α) (_ : p.last = a), p.length
```

Comment:
Right, I considered this. I think I stuck to this definition as it is closer to the prose description. Still worth changing?

Reply:
Whatever definition you choose, you can have the other one as a theorem. Best of both worlds!

Reply:
The definition with `p.last ≤ a` just seems more right to me to the extent that the corresponding picture is a increasing path below `a` (namely between `⊥` and `a`) rather than an increasing path ending right on `a` (but not necessarily starting at `⊥`).

As Violeta says, you can still have theorems that pretend that your current definition is still the definition

Reply:
Ok, will change back to that then.